server/cli/launch.py

```diff
@@ -389,36 +380,59 @@ def url_check(url):
     # Setup app
     cellxgene_url = f"http://{host}:{port}"
 
-    # Import Flask app
-    server = Server()
+    # app config
+    app_config = AppConfig(
+        title=title,
+        about=about,
+        scripts=scripts,
+        layout=embedding,
+        max_category_items=max_category_items,
+        diffexp_lfc_cutoff=diffexp_lfc_cutoff,
+        obs_names=obs_names,
+        var_names=var_names,
+        scanpy_backed=backed,
+        disable_diffexp=disable_diffexp,
+        dataroot=dataroot)
+
+    cxg_data = None
+    if matrix_data_loader:
+        try:
+            cxg_data = matrix_data_loader.open(app_config)
+        except RuntimeError as e:
+            raise click.ClickException(str(e))
+
+        if not disable_diffexp and cxg_data.parameters.get("diffexp_may_be_slow", False):
+            click.echo(
+                f"[cellxgene] CAUTION: due to the size of your dataset, "
+                f"running differential expression may take longer or fail."
+            )
 
-    server.create_app()
-    server.app.config.update(SCRIPTS=scripts)
+    # create an annotations object.  Only AnnotationsLocalFile is used (for now)
+    annotations = None
 
-    if not verbose:
-        log = logging.getLogger("werkzeug")
-        log.setLevel(logging.ERROR)
+    if experimental_annotations:
+        annotations = AnnotationsLocalFile(experimental_annotations_output_dir,
+                                           experimental_annotations_file)
 
-    file_size = data_locator.size() if data_locator.islocal() else 0
+        # if the user has specified a fixed label file, go ahead and validate it
+        # so that we can remove errors early in the process.
 
-    # if a big file, let the user know it may take a while to load.
-    if file_size > BIG_FILE_SIZE_THRESHOLD:
-        click.echo(f"[cellxgene] Loading data from {basename(data)}, this may take a while...")
-    else:
-        click.echo(f"[cellxgene] Loading data from {basename(data)}.")
+        if experimental_annotations_file:
+            cxg_data.validate_label_data(annotations.read_labels(cxg_data))
```

Comment:
this function does not exist in the branch.  It used to be in scanpy_engine

Reply:
I renamed the function and missed this use case.  fixed.